addons/mail/static/src/core/public_web/discuss.js

```diff
@@ -121,7 +120,7 @@ export class Discuss extends Component {
     }
 
     get thread() {
-        return this.store.discuss.thread;
+        return this.props.thread || this.store.discuss.thread;
```

Comment:
Cf. other comment, this should have a way to determine we are in livechat history but no thread was found.